### PR TITLE
Rename

### PR DESCRIPTION
**Pull request by oxinabox**
This is to change over to the new name

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
-name = "FDM"
-uuid = "e25cca7e-83ef-51fa-be6c-dfe2a3123128"
-version = "0.6.1"
+name = "FiniteDifferences"
+uuid = "8991a7d4-b31a-11e9-a2a3-2a2ae2dbcce4"
```
Comment (oxinabox) [inline Project.toml]:
```suggestion
uuid = "e25cca7e-83ef-51fa-be6c-dfe2a3123128"
```

Don't change the name

File: src/methods.jl [hunk 2]
```diff
@@ -48,10 +48,10 @@ mutable struct History
 end
 
 """
-    FDMethod
+    FiniteDifferencesethod
```
Comment (willtebbutt) [inline src/methods.jl]: Various instances of `FiniteDifferencesethod` should be `FiniteDifferenceMethod`

File: test/methods.jl [hunk 3]
```diff
@@ -25,9 +25,9 @@ using FDM: Forward, Backward, Central, Nonstandard
         @test central_fdm(5, 1)(abs, 0.001) ≈ 1.0
     end
 
-    @testset "Printing FDMethods" begin
+    @testset "Printing FiniteDifferencesethods" begin
```
Comment (willtebbutt) [inline test/methods.jl]: Similar, should be `FiniteDifferenceMethods`

File: src/methods.jl (hunk 2 above)
Comment (oxinabox) [reply src/methods.jl]: I th

```suggestion
    FiniteDifferenceMethod
```

File: src/methods.jl [hunk 4]
```diff
@@ -48,10 +48,10 @@ mutable struct History
...
 
 Abstract type for all finite differencing method types.
-Subtypes of `FDMethod` are callable with the signature
+Subtypes of `FiniteDifferencesethod` are callable with the signature
```
Comment (oxinabox) [inline src/methods.jl]:
```suggestion
Subtypes of `FiniteDifferenceMethod` are callable with the signature
```

File: src/methods.jl [hunk 5]
```diff
@@ -64,10 +64,10 @@ where the keyword arguments can be any of
 * `condition`: The condition number. See [`DEFAULT_CONDITION`](@ref).
 * `eps`: The assumed roundoff error. Defaults to `eps()` plus [`TINY`](@ref).
 """
-abstract type FDMethod end
+abstract type FiniteDifferencesethod end
```
Comment (oxinabox) [inline src/methods.jl]:
```suggestion
abstract type FiniteDifferenceMethod end
```

File: src/methods.jl [hunk 6]
```diff
@@ -64,10 +64,10 @@ where the keyword arguments can be any of
...
 
-function Base.show(io::IO, x::FDMethod)
-    @printf io "FDMethod:\n"
+function Base.show(io::IO, x::FiniteDifferencesethod)
```
Comment (oxinabox) [inline src/methods.jl]:
```suggestion
function Base.show(io::IO, x::FiniteDifferenceMethod)
```

File: src/methods.jl [hunk 7]
```diff
@@ -64,10 +64,10 @@ where the keyword arguments can be any of
...
+    @printf io "FiniteDifferencesethod:\n"
```
Comment (oxinabox) [inline src/methods.jl]:
```suggestion
    @printf io "FiniteDifferenceMethod:\n"
```

File: src/methods.jl [hunk 8]
```diff
@@ -83,7 +83,7 @@ end
 
 for D in (:Forward, :Backward, :Central, :Nonstandard)
     @eval begin
-        struct $D{G<:AbstractVector, C<:AbstractVector} <: FDMethod
+        struct $D{G<:AbstractVector, C<:AbstractVector} <: FiniteDifferencesethod
```
Comment (oxinabox) [inline src/methods.jl]:
```suggestion
        struct $D{G<:AbstractVector, C<:AbstractVector} <: FiniteDifferenceMethod
```

File: src/methods.jl [hunk 9]
```diff
@@ -116,24 +116,24 @@ for D in (:Forward, :Backward, :Central)
...
 
         @doc """
-            FDM.$($(Meta.quot(D)))(p, q; kwargs...)
+            FiniteDifferences.$($(Meta.quot(D)))(p, q; kwargs...)
             $($(Meta.quot(fdmf)))(p, q; kwargs...)
 
         Construct a $($lcname) finite difference method of order `p` to compute the `q`th
         derivative.
-        See [`FDMethod`](@ref) for more details.
+        See [`FiniteDifferencesethod`](@ref) for more details.
```
Comment (oxinabox) [inline src/methods.jl]:
```suggestion
        See [`FiniteDifferenceMethod`](@ref) for more details.
```

File: src/methods.jl [hunk 10]
```diff
@@ -116,24 +116,24 @@ for D in (:Forward, :Backward, :Central)
...
 
 """
-    FDM.Nonstandard(grid, q; kwargs...)
+    FiniteDifferences.Nonstandard(grid, q; kwargs...)
 
 An finite differencing method which is constructed based on a user-defined grid. It is
 nonstandard in the sense that it represents neither forward, backward, nor central
 differencing.
-See [`FDMethod`](@ref) for further details.
+See [`FiniteDifferencesethod`](@ref) for further details.
```
Comment (oxinabox) [inline src/methods.jl]:
```suggestion
See [`FiniteDifferenceMethod`](@ref) for further details.
```

File: src/methods.jl [hunk 11]
```diff
@@ -166,8 +166,8 @@ end
 _estimate_bound(x, cond) = cond * maximum(abs, x) + TINY
 
 """
-    fdm(m::FDMethod, f, x[, Val(false)]; kwargs...) -> Real
-    fdm(m::FDMethod, f, x, Val(true); kwargs...) -> Tuple{FDMethod, Real}
+    fdm(m::FiniteDifferencesethod, f, x[, Val(false)]; kwargs...) -> Real
+    fdm(m::FiniteDifferencesethod, f, x, Val(true); kwargs...) -> Tuple{FiniteDifferencesethod, Real}
```
Comment (oxinabox) [inline src/methods.jl]:
```suggestion
    fdm(m::FiniteDifferenceMethod, f, x, Val(true); kwargs...) -> Tuple{FiniteDifferenceMethod, Real}
```

File: src/methods.jl [hunk 12]
```diff
@@ -166,8 +166,8 @@ end
 _estimate_bound(x, cond) = cond * maximum(abs, x) + TINY
 
 """
-    fdm(m::FDMethod, f, x[, Val(false)]; kwargs...) -> Real
-    fdm(m::FDMethod, f, x, Val(true); kwargs...) -> Tuple{FDMethod, Real}
+    fdm(m::FiniteDifferencesethod, f, x[, Val(false)]; kwargs...) -> Real
```
Comment (oxinabox) [inline src/methods.jl]:
```suggestion
    fdm(m::FiniteDifferenceMethod, f, x[, Val(false)]; kwargs...) -> Real
```

File: src/methods.jl [hunk 13]
```diff
@@ -186,7 +186,7 @@ The recognized keywords are:
     `adapt` greater than 0 when `m::Nonstandard` results in an error.
 
 !!! note
-    Calling [`FDMethod`](@ref) objects is equivalent to passing them to `fdm`.
+    Calling [`FiniteDifferencesethod`](@ref) objects is equivalent to passing them to `fdm`.
```
Comment (oxinabox) [inline src/methods.jl]:
```suggestion
    Calling [`FiniteDifferenceMethod`](@ref) objects is equivalent to passing them to `fdm`.
```

File: src/methods.jl [hunk 14]
```diff
@@ -195,7 +195,7 @@ julia> fdm(central_fdm(5, 1), sin, 1; adapt=2)
 0.5403023058681039
 
 julia> fdm(central_fdm(2, 1), exp, 0, Val(true))
-(FDMethod:
+(FiniteDifferencesethod:
```
Comment (oxinabox) [inline src/methods.jl]:
```suggestion
(FiniteDifferenceMethod:
```

File: src/methods.jl [hunk 15]
```diff
@@ -217,7 +217,7 @@ function fdm(
     eps=(Base.eps(float(bound)) + TINY),
     adapt=m.history.adapt,
     max_step=0.1,
-) where M<:FDMethod
+) where M<:FiniteDifferencesethod
```
Comment (oxinabox) [inline src/methods.jl]:
```suggestion
) where M<:FiniteDifferenceMethod
```

File: src/methods.jl [hunk 16]
```diff
@@ -265,7 +265,7 @@ function fdm(
     return m, dfdx
 end
 
-function fdm(m::FDMethod, f, x, ::Val{false}=Val(false); kwargs...)
+function fdm(m::FiniteDifferencesethod, f, x, ::Val{false}=Val(false); kwargs...)
```
Comment (oxinabox) [inline src/methods.jl]:
```suggestion
function fdm(m::FiniteDifferenceMethod, f, x, ::Val{false}=Val(false); kwargs...)
```

File: test/methods.jl (hunk 3 above)
Comment (oxinabox) [inline test/methods.jl]:
```suggestion
    @testset "Printing FiniteDifferenceMethods" begin
```

File: test/methods.jl [hunk 17]
```diff
@@ -25,9 +25,9 @@ using FDM: Forward, Backward, Central, Nonstandard
...
         @test sprint(show, central_fdm(2, 1)) == """
-            FDMethod:
+            FiniteDifferencesethod:
```
Comment (oxinabox) [inline test/methods.jl]:
```suggestion
            FiniteDifferenceMethod:
```

File: test/methods.jl [hunk 18]
```diff
@@ -39,7 +39,7 @@ using FDM: Forward, Backward, Central, Nonstandard
         regex_array = r"\[([\d.+-e]+(, )?)+\]"
         @test occursin(Regex(join(map(x -> x.pattern,
             [
-                r"FDMethod:",
+                r"FiniteDifferencesethod:",
```
Comment (oxinabox) [inline test/methods.jl]:
```suggestion
                r"FiniteDifferenceMethod:",
```

**Pull request merged**